src/tests/suite/daos_obj.c

```diff
@@ -3365,6 +3365,169 @@ io_pool_map_refresh_trigger(void **state)
 	ioreq_fini(&req);
 }
 
+/**
+ * Test to fetch the non existence keys.
+ */
+static void nonexistent_internal(void **state, daos_obj_id_t oid,
+				 const char key_type[])
+{
+	test_arg_t	*arg = *state;
+	struct ioreq	req;
+	char		*update_buf;
+	char		*fetch_buf;
+	unsigned int	size   = IO_SIZE_NVME;
+
+	D_ALLOC(update_buf, size);
+	assert_non_null(update_buf);
+	dts_buf_render(update_buf, size);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+	D_ALLOC(fetch_buf, size);
+	assert_non_null(fetch_buf);
+	memset(fetch_buf, 0, size);
+
+	/* Insert single dkey/akey*/
+	insert_single("exist dkey", "exist akey", 0, update_buf, size,
+			DAOS_TX_NONE, &req);
+
+	if (strcmp(key_type, "akey") == 0) {
+		/*Fetch non existent akey*/
+		lookup_single("exist dkey", "nonexistent akey", 0, fetch_buf,
+			      size, DAOS_TX_NONE, &req);
+	} else if (strcmp(key_type, "dkey") == 0) {
+		/*Fetch non existent dkey*/
+		lookup_single("nonexistent dkey", "exist akey", 0, fetch_buf,
+			      size, DAOS_TX_NONE, &req);
+	} else if (strcmp(key_type, "oid") == 0) {
+		/*Fetch non existent oid*/
```

Comment:
This 'oid' is actually existing since there is an insert_single() call to this oid above. A different 'oid' should be used for this lookup_single() call.

Reply:
I will put a oid check for skipping the insert_single call on line 3389.

Reply:
then please don't forget to test nonexistent oid first, because dkey and akey test will 'create old' on server side.

Reply:
reviewed just before you pushed update, this works too, although you insert unnecessarily twice on dkey then akey check. anyway, not a big deal..

Reply:
ok will keep it as it is.